File: backend/src/main/java/by/kapitonov/computer/shop/backend/model/product/Motherboard.java
package by.kapitonov.computer.shop.backend.model.product;

import by.kapitonov.computer.shop.backend.model.*;
import by.kapitonov.computer.shop.backend.model.product.detail.MotherboardFormFactor;
import by.kapitonov.computer.shop.backend.model.product.detail.Socket;
import by.kapitonov.computer.shop.backend.model.product.detail.RamMemoryType;
import lombok.*;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.*;

@Entity
@Table(name = "motherboards")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EqualsAndHashCode
@ToString
@EntityListeners(AuditingEntityListener.class)
public class Motherboard extends AuditEntity {


    @Column(name = "number_of_memory_slot", nullable = false)
    private Integer numberOfMemorySlot;

    @Column(name = "maximum_memory_size", nullable = false)
    private Integer maximumMemorySize;

    @Column(name = "maximum_memory_frequency", nullable = false)
    private Integer maximumMemoryFrequency;

    @OneToOne(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id", referencedColumnName = "id")
    private Product product;

    @OneToOne
    @JoinColumn(name = "socket_id", referencedColumnName = "id")
    private Socket socket;

    @OneToOne
    @JoinColumn(name = "form_factor_id", referencedColumnName = "id")
    private MotherboardFormFactor formFactor;

    @OneToOne
    @JoinColumn(name = "type_of_memory_id", referencedColumnName = "id")
    private RamMemoryType typeOfMemory;


}

File: frontend/src/app/model/Motherboard.ts
import {Product} from "./Product";

export class Motherboard {

  private _id: number;
  private _numberOfMemorySlot: number;
  private _maximumMemorySize: number;
  private _maximumMemoryFrequency: number;
  private _socket: string;
  private _formFactor: string;
  private _typeOfMemory: string;
  private _product: Product;

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get numberOfMemorySlot(): number {
    return this._numberOfMemorySlot;
  }

  set numberOfMemorySlot(value: number) {
    this._numberOfMemorySlot = value;
  }

  get maximumMemorySize(): number {
    return this._maximumMemorySize;
  }

  set maximumMemorySize(value: number) {
    this._maximumMemorySize = value;
  }

  get maximumMemoryFrequency(): number {
    return this._maximumMemoryFrequency;
  }

  set maximumMemoryFrequency(value: number) {
    this._maximumMemoryFrequency = value;
  }

  get socket(): string {
    return this._socket;
  }

  set socket(value: string) {
    this._socket = value;
  }

  get formFactor(): string {
    return this._formFactor;
  }

  set formFactor(value: string) {
    this._formFactor = value;
  }

  get typeOfMemory(): string {
    return this._typeOfMemory;
  }

  set typeOfMemory(value: string) {
    this._typeOfMemory = value;
  }

  get product(): Product {
    return this._product;
  }

  set product(value: Product) {
    this._product = value;
  }
}
